File: microservice-architecture/configuration-center/js/test/test_configuration_center.js
import { ConfigCenter, ConfigClient } from '../src/configuration_center.js'

function assertEqual(expected, actual, message) {
  if (expected !== actual) {
    throw new Error(`${message} expected=${expected} actual=${actual}`)
  }
}

const center = new ConfigCenter()
center.put({
  serviceName: 'order-service',
  environment: 'prod',
  version: 1,
  dbHost: 'db.prod.internal',
  timeoutMs: 300,
  featureOrderAudit: false
})

const client = new ConfigClient(center, 'order-service', 'prod')
const loaded = client.load()
assertEqual(1, loaded.version, 'initial version')
assertEqual(300, loaded.timeoutMs, 'initial timeout')

center.put({
  serviceName: 'order-service',
  environment: 'prod',
  version: 2,
  dbHost: 'db.prod.internal',
  timeoutMs: 500,
  featureOrderAudit: true
})

const refreshed = client.refresh()
assertEqual(2, refreshed.version, 'refreshed version')
assertEqual(true, refreshed.featureOrderAudit, 'feature flag')

console.log('configuration-center(js) tests passed')

File: microservice-architecture/configuration-center/python/test/test_configuration_center.py
import os
import sys
import unittest

sys.path.append(os.getcwd())

from src.configuration_center import ConfigCenter, ConfigClient, ServiceConfig  # noqa: E402


class ConfigurationCenterTest(unittest.TestCase):
    def test_load_and_refresh(self) -> None:
        center = ConfigCenter()
        center.put(ServiceConfig("order-service", "prod", 1, "db.prod.internal", 300, False))

        client = ConfigClient(center, "order-service", "prod")
        loaded = client.load()
        self.assertIsNotNone(loaded)
        self.assertEqual(1, loaded.version)
        self.assertEqual(300, loaded.timeout_ms)

        center.put(ServiceConfig("order-service", "prod", 2, "db.prod.internal", 500, True))
        refreshed = client.refresh()
        self.assertIsNotNone(refreshed)
        self.assertEqual(2, refreshed.version)
        self.assertTrue(refreshed.feature_order_audit)


if __name__ == "__main__":
    unittest.main()
